.github/actions/pre_make/action.yml

```diff
@@ -0,0 +1,44 @@
+name: "Pre-Make"
+runs:
+  using: composite
+  steps:
+    - name: Restore make from cache
+      id: cache-make
+      uses: actions/cache@v3
+      with:
+        path: make.bin
+        key: v13_make.bin-${{ hashFiles('**/make.md5') }}
+    - name: Install make
+      if: steps.cache-make.outputs.cache-hit != 'true'
+      run: |
+        if [ -s make.bin ] ; then
+          echo "make.bin exists; not installing"
+        else
+          echo "/usr/bin/make doesn't exist; installing"
+          sudo apt-get update
+          sudo apt-get install -y make
```

Comment:
nit: This is pretty inconsequential, but these days, `apt-get` is the lower level package manager on debian based distributions, and `apt` is the higher level one that's generally used by end users, so we should be able to swap out  `apt` in every place we're currently using `apt-get`.